### ISSUE TITLE
401 Error using C# API from Console App

### ISSUE THREAD

**Issue by vinthefantastic1** [tripit/api#194]
I'm using the tripit provided C# app template.

I use my developer key and secret to  get the initial token/secret.  That works fine.

However after receiving the initial token and secret, and trying to use them to get the authorizedToken, I get the 401 error.

What am I missing?

`class Program
    {
        static string apiUrl = "https://api.tripit.com";
        static void Main(string[] args)
        {

            string consumerKey = "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx";
            string consumerSecret = "dxxxxxxxxxxxxxxxxxxxxxxx";

            OAuthCredential initialCredential = getToken(consumerKey, consumerSecret);

            OAuthCredential authorizedCredential = getAuthorizedToken(consumerKey, consumerSecret, initialCredential.Token, initialCredential.TokenSecret);

            Console.WriteLine("token: {0} secret:{1}", authorizedCredential.Token, authorizedCredential.TokenSecret);

            Credential credential = new OAuthCredential(
                consumerKey, consumerSecret,
                authorizedCredential.Token, authorizedCredential.TokenSecret);

            TripIt t = new TripIt(credential, apiUrl);

            Console.WriteLine(t.ListTrip().Trip.Length);
            Console.ReadLine();

        }

        // get a basic token to in turn ask for an authorized credential...
        static OAuthCredential getToken(string consumerKey, string consumerSecret)
        {

            Credential credential = new OAuthCredential(
                consumerKey, consumerSecret);

            TripIt t = new TripIt(credential, apiUrl);

            OAuthCredential authorizedCredential = t.GetRequestToken();

            Console.WriteLine("Request Token: " + authorizedCredential.Token);
            Console.WriteLine("Token Secret:  " + authorizedCredential.TokenSecret);

            return authorizedCredential;

        }

        static OAuthCredential getAuthorizedToken(string consumerKey, string consumerSecret, string requestToken, string requestTokenSecret)
        {

           Credential credential = new OAuthCredential(
                consumerKey, consumerSecret,
                requestToken, requestTokenSecret);

            TripIt t2 = new TripIt(credential, apiUrl);

            OAuthCredential authorizedCredential = t2.GetAccessToken();

            Console.WriteLine("Authorized Token: " + authorizedCredential.Token);
            Console.WriteLine("Token Secret:     " + authorizedCredential.TokenSecret);

            return authorizedCredential;

        }

    }`

**Comment by igordossantos** [comment]
Were you able to resolve this issue?

**Comment by vinthefantastic1** [comment]
No... I just switched to basic authentication for my purposes. Don't use basic auth in a production app though.

**Comment by MrHinsh** [comment]
Do you have a basic auth example, Im having the same issue.
